client/components/ReviewList.jsx

```diff
@@ -1,29 +1,84 @@
+/* eslint-disable no-lonely-if */
 /* eslint-disable react/forbid-prop-types */
 /* eslint-disable react/require-default-props */
 /* eslint-disable no-underscore-dangle */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import ReviewListEntry from './ReviewListEntry';
+import ReviewListPageNav from './ReviewListPageNav';
 
-const ReviewList = ({ reviews, handleClick }) => (
-  <div>
-    {reviews.map((review) => (
-      <ReviewListEntry
-        handleClick={handleClick}
-        key={review._id}
-        name={review.name}
-        rating={review.rating}
-        date={review.date}
-        review={review.review}
-        attachedPic={review.attached_pic}
-        profilePic={review.profile_pic}
-        purchasedItemName={review.purchased_item_name}
-        purchasedItemPic={review.purchased_item_pic}
-        reviewData={review}
+const ReviewList = ({ reviews, handleClick }) => {
+  const [reviewPages, setReviewPages] = useState([]);
+  const [totalPages, setTotalPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(0);
+
+  const splitReviews = () => {
+    const results = [];
+    const array = [...reviews];
+
+    while (array.length > 0) {
+      results.push(array.splice(0, 4));
+    }
+
+    return results;
+  };
+
+  useEffect(() => {
+    // split reviews array into arrays of 4;
+    setReviewPages(splitReviews());
+
+    // set the total number of pages
+    setTotalPages(Math.ceil(reviews.length / 4));
+  }, [reviews]);
```

Comment:
You might consider using an if statement here instead of useEffect.  It doesn't look like there'd ever be a case where you wouldn't have your reviews data on the initial render, so an if statement checking the see if the reviews length is greater than 4 should let you run both functions without rendering first and then re-rendering once you're finished.

Reply:
You could also consider pulling the function definition for splitReviews() out of the component and just initializing reviewPages with the result of splitReviews.  I'm not sure that the benefits outweigh the effort/possible readability hit, but... it is an option!

Reply:
Implemented!